scripts/lib/libvirt/libvirt_setup.py

```diff
@@ -202,15 +225,11 @@ def cleanup(args):
     for dom in domains:
         domain_cleanup(dom)
 
-    networks = [i for i in conn.listAllNetworks()
-                if i.name() == "{0}-admin".format(args.cloud)]
-    for network in networks:
-        if network.isActive():
-            print("destroying {0}".format(network.name()))
+    for network in conn.listAllNetworks():
+        if network.isActive() and network.name() == args.cloud + "-admin":
+            print("Cleaning up network {0}".format(network.name()))
             network.destroy()
-
-        print("undefining {0}".format(network.name()))
-        network.undefine()
+            network.undefine()
```

Comment:
The old code was undefining the network even if it wasn't active, which makes sense (in case it was manually destroyed). Could you keep that?